Add error.hint semantic convention (#822)
open-telemetry/opentelemetry-specification: specification/trace/semantic_conventions/span-general.md

| Ruby                  |                                        | `Thread.current.name`              |
| C++                   | `std::this_thread::get_id()`             |                                    |
| Erlang               | `erlang:system_info(scheduler_id)` |                                  |

## Error Attributes

A span or event can be annotated with an `error.hint` attribute to indicate that
an error condition was observed.

> When recording an [exception event](exceptions.md) and the exception is
> considered to be an error, both the event and the span SHOULD have matching
> `error.hint` attributes set to `true`. It's possible for more than one exception

Reviewer: ```suggestion
If there are [exception events](exceptions.md) on a span, the span attribute `error.hint`
SHOULD be set to `true`, if at least one of the event's `error.hint` attribute is `true`.
It's possible for more than one exception
```
I think this explains the case of multiple events with potentially different hints better.

event to have an `error.hint` attribute set. An `error.hint` represents an
instrumentation author's best judgement as to whether or not an error condition

> was encountered, but the decision is not definitive. Tracing backends can use

Reviewer 2: >but the decision is not definitive

I think this phrase does not add much value. I think this paragraph works better without it.

the `error.hint` attribute as a cue for further analysis and can ultimately
choose to honor or ignore the hint.

| Attribute name | Notes and examples                                                                                                       |
| :------------- | :----------------------------------------------------------------------------------------------------------------------- |

> | error.hint     | A boolean value that represents an instrumentation author's best judgement as to whether an error condition was observed |

Reviewer: ```suggestion
| error.hint     | A boolean value that represents an instrumentation author's best judgement as to whether an error condition was observed. |
```
